File: practica1/P1S3/FilterManager.cpp

#pragma once
#include "Filtro.cpp"
#include "Filtro1.cpp"
#include "Filtro2.cpp"
#include "FilterChain.cpp"

class FilterManager
{
    private:
        FilterChain cadena;

    public:
        FilterManager(){}

        FilterManager(FilterChain c)
        {
            cadena = c;
        }

        void aniadirFiltro(Filtro *f)
        {
            cadena.aniadirFiltro(f);
        }

        void ejecutarFiltros()
        {
            cadena.ejecutarFiltros();
        }

        void vaciarFiltro()
        {
            cadena.vaciarFiltro();
        }
};

File: practica1/P1S3/Java/src/practica1dss3/FilterManager.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package practica1dss3;

/**
 *
 * @author USUARIO
 */
public class FilterManager {
    private FilterChain cadena;
    
    public FilterManager(FilterChain c){
        cadena = c;
    }
    
    public void añadirFiltro(Filtro f){
        cadena.añadirFiltro(f);
    }
    
    public void ejecutarFiltros(){
        cadena.ejecutarFiltros();
    }
    
    public void vaciarFiltros(){
        cadena.vaciarFiltros();
    }
}
